share/wrf_tsin.F

```diff
@@ -54,9 +56,22 @@ SUBROUTINE wrf_tsin ( grid , ierr )
                 ! Read in time series locations
                 istatus = 0
                 DO WHILE (istatus == 0)
-                   READ(UNIT=iunit, FMT='(A25,1X,A5,1X,F7.3,1X,F8.3)', IOSTAT=istatus)            &
-                        grid%desctsloc(grid%ntsloc+1), grid%nametsloc(grid%ntsloc+1), &
-                        grid%lattsloc(grid%ntsloc+1), grid%lontsloc(grid%ntsloc+1)
+                   ! if tslist is ij, read in integers and not floats
+                   IF (config_flags%map_proj == 0) THEN !ideal run will only use i,j
+                       READ(UNIT=iunit, FMT='(A25,1X,A5,1X,I9,1X,I10)', IOSTAT=istatus)            &
+                       grid%desctsloc(grid%ntsloc+1), grid%nametsloc(grid%ntsloc+1), &
+                       grid%itsloc(grid%ntsloc+1), grid%jtsloc(grid%ntsloc+1)
+                   ELSE
+                       IF (config_flags%tslist_ij) THEN
```

Comment:
To avoid repeating three dense lines of reading code, we could combine this if-test with the one on `map_proj` so that the code block looks like this:
```fortran
                   ! If this is an idealized case (with no map projection) or if coordinates are
                   ! specified as (i,j) read into itsloc and jtsloc; otherwise, read into lattsloc
                   ! and lontsloc.
                   IF ( (config_flags%map_proj == 0) .OR. config_flags%tslist_ij ) THEN 
                       READ(UNIT=iunit, FMT='(A25,1X,A5,1X,I9,1X,I10)', IOSTAT=istatus) &
                       grid%desctsloc(grid%ntsloc+1), grid%nametsloc(grid%ntsloc+1),    &
                       grid%itsloc(grid%ntsloc+1), grid%jtsloc(grid%ntsloc+1)
                   ELSE
                       READ(UNIT=iunit, FMT='(A25,1X,A5,1X,F7.3,1X,F8.3)', IOSTAT=istatus) &  
                       grid%desctsloc(grid%ntsloc+1), grid%nametsloc(grid%ntsloc+1),       &
                       grid%lattsloc(grid%ntsloc+1), grid%lontsloc(grid%ntsloc+1)
                   END IF
```